tools/preprocessor.mjs

```diff
@@ -31,11 +31,17 @@ Usage: preprocessor.mjs <settings.json> <input-file> [--expand-macros]`);
   process.exit(0);
 }
 
+loadDefaultSettings();
+
 assert(positionals.length == 2, 'Script requires 2 arguments');
+
+// Load settings from JSON passed on the command line
 const settingsFile = positionals[0];
-const inputFile = positionals[1];
+assert(settingsFile, 'settings file not specified');
+const user_settings = JSON.parse(readFile(settingsFile));
```

Comment:
```suggestion
const userSettings = JSON.parse(readFile(settingsFile));
```

Reply:
Done.  I also updated compiler.mjs where this code was copied from.